pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-http-request.adoc

```diff
@@ -1,55 +1,70 @@
-= Perform an HTTP Call
+= Performing an HTTP Call
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images
 
-To interact with external services, the policy must use HTTP calls. You can perform HTTP calls in the request and response filters and in the function. To perform an HTTP call, inject the `HttpClient` as follows:
+To interact with external services, the policy must use HTTP calls. You can perform HTTP calls in the `[entrypoint]` function, the request filter, or the response filter. To perform an HTTP call, inject the `HttpClient` as follows:
 
-* Injection for `on_request` or `on_response` wrapped functions functions:
+* Injection for `[entrypoint]` function:
 +
 [source,Rust]
 ----
-​​async fn request_filter(state: RequestState, conf: &Config, client: HttpClient) {
- ...
-}
+#[entrypoint]
+async fn configure(
+    launcher: Launcher,
+    Configuration(configuration): Configuration,
+    client: HttpClient,
+) -> Result<(), LaunchError> {
 ----
 
-* Injection for `[entrypoint]` function:
+* Injection for `on_request` or `on_response` wrapped functions functions:
 +
 [source,Rust]
 ----
+// Handler with the logic for incoming requests
 ​​async fn request_filter(state: RequestState, conf: &Config, client: HttpClient) {
  ...
 }
-----
 
-You can only make HTTP calls to services defined in the Flex Gateway configuration files. For more information about defining services, see xref:flex-local-configuration-reference-guide.adoc#service[Services].
+#[entrypoint]
+async fn configure(launcher: Launcher, Configuration(bytes): Configuration) -> Result<()> {
+    let config: Config = serde_json::from_slice(&bytes).unwrap();
 
-For both Connected and Local modes, define services using the `Service` resource. If the service is not defined when the request is made, the API request fails.
+    // For incoming request events, using the on_request wrapper, inject the client and pass it to the request_filter handler
+    launcher
+        .launch(on_request(|request, client| {
+            request_filter(request, &config, client)
+        }))
+        .await?;
+    Ok(())
+}
+----
 
-For an example of how to make HTTP calls from the policy, given the `Service` resource:
+Define your service as a parameter in you schema definition to reuse your policy with different services. The following code snippet provides an example schema definition for an external service:
```

Comment:
This is not very accurate. The definition is a *must* to be able to make HTTP calls, not just so you can reuse it with different services.